cf_units/__init__.py

```diff
@@ -406,17 +406,33 @@ def _discard_microsecond(date):
     dates = np.asarray(date)
     shape = dates.shape
     dates = dates.ravel()
+    result = None
     # Create date objects of the same type returned by cftime.num2date()
     # (either datetime.datetime or cftime.datetime), discarding the
     # microseconds
-    dates = np.array(
-        [
-            d
-            and d.__class__(d.year, d.month, d.day, d.hour, d.minute, d.second)
-            for d in dates
-        ]
-    )
-    result = dates[0] if shape == () else dates.reshape(shape)
+    discard = []
+    for dt in dates:
+        if dt:
```

Comment:
I've honoured this short-circuit logic here, although I must confess I don't completely appreciate why we have to do this... I'm sure it's historical.